smack-core/src/main/java/org/jivesoftware/smack/packet/AbstractStreamOpen.java

```diff
@@ -0,0 +1,81 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.packet;
+
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.packet.StreamOpen.StreamContentNamespace;
+import org.jivesoftware.smack.util.StringUtils;
+
+/**
+ * AbstractStreamOpen is actually a {@link TopLevelStreamElement}, however we
+ * implement {@link Nonza} here. This is because, {@link XMPPConnection} doesn't
+ * yet support sending {@link TopLevelStreamElement} directly and the same can only
+ * be achieved through {@link XMPPConnection#sendNonza(Nonza)}.
+ */
+public abstract class AbstractStreamOpen implements Nonza {
+    public static final String ELEMENT = "stream:stream";
```

Comment:
That is a property of the concrete subclass, i.e., `StreamOpen`, not the abstract superclass.